sqlserver/datadog_checks/sqlserver/sqlserver.py

```diff
@@ -168,28 +168,21 @@ def __init__(self, name, init_config, instances):
             )
         if is_affirmative(self.instance.get('include_fci_metrics', False)):
             check_queries.extend([QUERY_FAILOVER_CLUSTER_INSTANCE])
-        self._check_queries = QueryExecutor(
-            self.execute_query_raw,
-            self,
-            queries=check_queries,
-            tags=self.tags,
-            hostname=self.resolved_hostname,
-        )
+        self._check_queries = self._new_query_executor(check_queries)
         self.check_initializations.append(self._check_queries.compile_queries)
 
-        self.server_state_queries = QueryExecutor(
-            self.execute_query_raw,
-            self,
-            queries=[QUERY_SERVER_STATIC_INFO],
-            tags=self.tags,
-            hostname=self.resolved_hostname,
-        )
+        self.server_state_queries = self._new_query_executor([QUERY_SERVER_STATIC_INFO])
         self.check_initializations.append(self.server_state_queries.compile_queries)
 
         # use QueryManager to process custom queries
         self._query_manager = QueryManager(
             self, self.execute_query_raw, tags=self.tags, hostname=self.resolved_hostname
         )
+
+        # this query executor must be initialized after the connection is made
+        # as the query can vary depending on the state of the database
+        self.file_stats_queries = None
```

Comment:
I think this would be cleaner as a lazy-loaded attribute:

```python
@property
def file_stats_queries():
    ...
```

Reply:
Good point, updated